libqtile/layout/floating.py

```diff
@@ -231,12 +249,6 @@ def configure(self, client, screen_rect):
             client.user_placed_window_setup(bc, bw)
             return
 
-        # similar to above but the X11 version, the client may have already
-        # placed itself. let's respect that
-        if client.has_user_set_position():
-            client.user_placed_window_setup(bc, bw)
```

Comment:
Is there a way we can avoid a place() call if has_user_set_position() is true and we don't have to change the coordinates? It does avoid some flicker.

Reply:
Not calling `_Window.place()` seems to be what is causing #1893. I guess that means that the call to `_Window.send_configure_notify`is the important part? Other than that `place` does what `user_placed_window_setup` does, as far as I can see. I'll try to take a closer look.

@m-col : Your PR #1887 touches a lot of this code, too. I'll try to see whether that, too, solves #1893.

Reply:
It seems that #1887 also solves #1893. It doesn't adress the cases where floating windows spawn on the wrong screen, however. So I think #1887 would be preferrable over this PR, I could then later try to fix the problem with spawning on the wrong screen separately.

Reply:
I put some time into investigating why calling `place` for windows having `USPosition` and/or `PPosition` solves having invisible popups/children (#1893): The critical call needed is that to `xcbq.Window.configure(x, y, width, height)`.

This seems very confusing to me, since the window should already be positioned and visible without that call, right? For the cases described in #1893, however, only _every other_ instance appears visible. There might be an underlying bug in how we grab and manage windows, but I don't think I can find it on my own.

Here is the change that would fix #1893 - maybe somebody else knows why this works and what underlying issue it might be pointing to? (@m-col , @ramnes ?)
```python
# inside layout.floating.Floating.configure
if client.has_user_set_position():
    client.user_placed_window_setup(bc, bw)
    client.window.configure(x=client.x, y=client.y, width=client.width, height=client.height)
```

Any help would be very much appreciated!